source/Applications/Advanced/HandEyeCalibration/UtilizeEyeInHandCalibration/UtilizeEyeInHandCalibration.cs

```diff
@@ -0,0 +1,159 @@
+
+                    Vector<float> pointInCameraFrame = CreateVector.DenseOfArray(pickingPoint);
+                    Console.WriteLine("Point coordinates in camera frame: " + pointInCameraFrame);
+
+                    Console.WriteLine("Transforming (picking) point from camera to robot base frame");
+                    Vector<float> pointInBaseFrame = transformBaseToCamera * pointInCameraFrame;
+
+                    Console.WriteLine("Point coordinates in robot base frame: " + pointInBaseFrame);
+                    break;
+
+                case CommandType.TransformPointCloud:
+
+                    Console.WriteLine("Transforming point cloud");
+                    var transformMatrix = mathDotNetToZivid(transformBaseToCamera);
+                    pointCloud.Transform(transformMatrix);
+
+                    var dataFileTransformed = "ZividGemTransformed.zdf";
+                    Console.WriteLine("Saving frame to file: " + dataFileTransformed);
+                    frame.Save(dataFileTransformed);
+                    break;
+
+                case CommandType.Unknown:
+                    Console.WriteLine("Error: Unknown command");
+                    break;
+            }
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine("Error: {0}", ex.Message);
+            Environment.ExitCode = 1;
+        }
+    }
+    static float[,] getTransformationMatrixFromYAML(string transformFile)
+    {
+        using (var reader = new StreamReader(transformFile))
+        {
+            // OpenCV writes yaml 1.0 with legacy first line
+            // YamlDotNet does not support this legacy first line
+            // so to work around we modify the first line if it is legacy
+            var first_line = reader.ReadLine();
+            if (first_line.Contains("%YAML:1.0"))
+            {
+                first_line = first_line.Replace(":", " ");
+            }
+            var new_reader = new StringReader(first_line + "\n" + reader.ReadToEnd());
+
+            var yaml = new YamlStream();
+            yaml.Load(new_reader);
+
+            var poseStateNode = (YamlMappingNode)yaml.Documents[0].RootNode;
+            var poseStateData = poseStateNode["PoseState"]["data"].ToString();
+            var matrixAsString = poseStateData.Trim('[', ']').Split(',');
+
+            int rows = int.Parse(poseStateNode["PoseState"]["rows"].ToString());
+            int cols = int.Parse(poseStateNode["PoseState"]["cols"].ToString());
+
+            float[,] zividMatrix = new float[rows, cols];
+
+            for (int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < cols; j++)
+                {
+                    zividMatrix[i, j] = float.Parse(matrixAsString[i * rows + j]);
+                }
+            }
+            return zividMatrix;
+        }
+    }
+    static Matrix<float> zividToMathDotNet(float[,] zividMatrix)
```

Comment:
I really like how you introduce named functions for these very simple one-liners. It makes the code which uses them easier to understand.

Reply:
Thanks!